File: java/unidad_02/Problema_2_06/src/Problema_2_06.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * El programa, dado como dato un número entero, determina si el
 * mismo es par, impar o nulo.
 */

public class Problema_2_06 {

    public static void main(String[] args) {

        int numero;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("Número entero: ");
            numero = scanner.nextInt();

        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        String mensaje = (Math.pow(-1, numero) > 0) ? "número par" : "número impar";
        // -1 elevado a un número par, es positivo
        // -1 elevado a un número impar, es negativo

        System.out.printf("%d es un %s", numero, mensaje);
    }
}

File: python/unidad_02/problema_2_06.py
"""
El programa, dado como dato un número entero, determina si el
mismo es par, impar o nulo.
"""

def main():

    try:

        numero = int(input("Número entero: "))

    except ValueError as e:

        print(e)
        return       
    
    mensaje = "número par" if (- 1) ** numero > 0 else "número impar"
    # -1 elevado a un número par, es positivo
    # -1 elevado a un número impar, es negativo

    print(f"{numero} es un {mensaje}")

if __name__ == '__main__':

    main()